scipy/interpolate/_interpolate.py

```diff
@@ -2452,10 +2463,18 @@ class RegularGridInterpolator:
     # this class is based on code originally programmed by Johannes Buchner,
     # see https://github.com/JohannesBuchner/regulargrid
 
+    SPLINE_DEGREE_MAP = {"slinear": 1, "cubic": 3, "quintic": 5, }
+    SPLINE_METHODS = list(SPLINE_DEGREE_MAP.keys())
+    ALL_METHODS = ["linear", "nearest"] + SPLINE_METHODS
+
     def __init__(self, points, values, method="linear", bounds_error=True,
                  fill_value=np.nan):
-        if method not in ["linear", "nearest"]:
+        if method not in self.ALL_METHODS:
             raise ValueError("Method '%s' is not defined" % method)
+        elif method in self.SPLINE_METHODS:
+            self._validate_not_complex(points, method)
```

Comment:
This line tries to  create a ragged array for e.g. `x, y = np.arange(5), np.arange(8)` and `RGI((x, y), values, method='cubic')`. Actually, why checking this at all? We should either check it for all methods or none at all. Spline-based methods will check it at the `__call__` time (via `make_interp_spline`), nearest and linear would probably do some nonsense.

Reply:
Maybe I thought these methods do not support complex inputs. But if these methods can support these, I agree these should be removed.

Reply:
Complex-valued `points`, no, nothing supports that AFAIK.

Reply:
Oh, OK. Complex-valued `values` is supported, but complex-valued `points` are not supported. I see.

Reply:
Yes, I think complex-valued N-dim vector spaces are best left out. A `C^N` space is equivalent to a `R^{2N}` space anyway, so users who really needs these can DIY it.